source/codegen/templates/proto.mako

```diff
@@ -49,6 +49,9 @@ service ${service_class_prefix} {
 ${mako_helper.define_attribute_enum(group.name, define_attribute_enum, attributes, config)}\
 %   endfor
 % endfor
+% for name, bitfield_enum in common_helpers.get_bitfield_enum_groups(data).items():
+${mako_helper.define_bitfield_enum(name, bitfield_enum)}\
+%endfor
```

Comment:
I don't think we need to add entirely new template code for these enum values. `GetDeviceAttributeInt32` and `GetPhysicalChanAttributeInt32` already have function enums.

I think there are two ways to update the existing codegen:

**A. Generate only the bitfield enum values**
```diff
diff --git a/source/codegen/common_helpers.py b/source/codegen/common_helpers.py
index 38185e38..c283d444 100644
--- a/source/codegen/common_helpers.py
+++ b/source/codegen/common_helpers.py
@@ -501,6 +501,10 @@ def get_attribute_enums_by_type(attributes):
         if "enum" in attribute:
             attribute_type = get_underlying_type(attribute)
             enum_name = attribute["enum"]
+            if "bitfield_enum" in attribute:
+                enum_name = attribute["bitfield_enum"]
+                if enum_name.startswith("_"):  # nidaqmx-python uses a leading underscore
+                    enum_name = enum_name[1:]
             attribute_enums_by_type[attribute_type].add(enum_name)
     return attribute_enums_by_type
```

This will delete 
```
  DEVICE_INT32_COUPLING1_AC = 10045;
  DEVICE_INT32_COUPLING1_DC = 10050;
  DEVICE_INT32_COUPLING1_GND = 10066;
```
and add
```
  DEVICE_INT32_COUPLING_TYPES_AC = 1;
  DEVICE_INT32_COUPLING_TYPES_DC = 2;
  DEVICE_INT32_COUPLING_TYPES_GROUND = 4;
  DEVICE_INT32_COUPLING_TYPES_HF_REJECT = 8;
  DEVICE_INT32_COUPLING_TYPES_LF_REJECT = 16;
  DEVICE_INT32_COUPLING_TYPES_NOISE_REJECT = 32;
```
Removing `DEVICE_INT32_COUPLING1_AC` breaks compatibility, but I don't think it is a useful enum value.

Also, when some attributes use the enum and other attributes use the bitfield_enum, it generates both:
```
  DEVICE_INT32_TRIGGER_USAGE_ADVANCE = 12488;
  DEVICE_INT32_TRIGGER_USAGE_PAUSE = 12489;
  DEVICE_INT32_TRIGGER_USAGE_REFERENCE = 12490;
  DEVICE_INT32_TRIGGER_USAGE_START = 12491;
  DEVICE_INT32_TRIGGER_USAGE_HANDSHAKE = 10389;
  DEVICE_INT32_TRIGGER_USAGE_ARM_START = 14641;
  DEVICE_INT32_TRIGGER_USAGE_TYPES_ADVANCE = 1;
  DEVICE_INT32_TRIGGER_USAGE_TYPES_PAUSE = 2;
  DEVICE_INT32_TRIGGER_USAGE_TYPES_REFERENCE = 4;
  DEVICE_INT32_TRIGGER_USAGE_TYPES_START = 8;
  DEVICE_INT32_TRIGGER_USAGE_TYPES_HANDSHAKE = 16;
  DEVICE_INT32_TRIGGER_USAGE_TYPES_ARM_START = 32;
```
However, we only do that for one of the navigation attributes:
```
        12324: {
            'access': 'read',
            'c_function_name': 'DevNavTrigUsage',
            'calling_convention': 'StdCall',
            'ctypes_data_type': 'numpy.int32',
            'enum': 'TriggerUsage',
            'handle_parameters': {
                'deviceHandle': {
                    'accessor': 'self._name',
                    'ctypes_data_type': 'ctypes.c_char_p',
                    'cvi_name': 'deviceName'
                }
            },
            'has_explicit_read_buffer_size': True,
            'has_explicit_write_buffer_size': True,
            'is_list': True,
            'is_python_object': False,
            'name': 'NAV_TRIG_USAGE',
            'python_data_type': 'TriggerUsage',
            'python_description': 'Indicates the trigger types supported by the device for a given measurement type.',
            'resettable': False,
            'type': 'int32[]'
        },
```

**B. Generate both the enum and bitfield enum values**
```diff
diff --git a/source/codegen/common_helpers.py b/source/codegen/common_helpers.py
index 38185e38..a61dae03 100644
--- a/source/codegen/common_helpers.py
+++ b/source/codegen/common_helpers.py
@@ -502,6 +502,12 @@ def get_attribute_enums_by_type(attributes):
             attribute_type = get_underlying_type(attribute)
             enum_name = attribute["enum"]
             attribute_enums_by_type[attribute_type].add(enum_name)
+        if "bitfield_enum" in attribute:
+            attribute_type = get_underlying_type(attribute)
+            enum_name = attribute["bitfield_enum"]
+            if enum_name.startswith("_"):  # nidaqmx-python uses a leading underscore
+                enum_name = enum_name[1:]
+            attribute_enums_by_type[attribute_type].add(enum_name)
     return attribute_enums_by_type
```
This will generate both:
```
  DEVICE_INT32_COUPLING1_AC = 10045;
  DEVICE_INT32_COUPLING1_DC = 10050;
  DEVICE_INT32_COUPLING1_GND = 10066;
  DEVICE_INT32_COUPLING_TYPES_AC = 1;
  DEVICE_INT32_COUPLING_TYPES_DC = 2;
  DEVICE_INT32_COUPLING_TYPES_GROUND = 4;
  DEVICE_INT32_COUPLING_TYPES_HF_REJECT = 8;
  DEVICE_INT32_COUPLING_TYPES_LF_REJECT = 16;
  DEVICE_INT32_COUPLING_TYPES_NOISE_REJECT = 32;
```

@astarche @reckenro @zhindes Anyone opposed to option A?

Reply:
I have implemented the changes based on option A.

Reply:
I think option A makes sense and I agree it doesn't seem like the old enum values like ` DEVICE_INT32_COUPLING1_AC` were that useful.